src/Viewer.tsx

```diff
@@ -84,9 +84,12 @@ function Viewer(): ReactElement {
 
   const [, startTransition] = React.useTransition();
 
-  const canv = useConstructor(() => {
+  // TODO: Make `canv` an IRenderCanvas type? Don't wrap `ColorizeCanvas3D` in `CanvasOverlay`...?
+  const canv: IRenderCanvas = useConstructor(() => {
     const stateDeps = renderCanvasStateParamsSelector(useViewerStateStore.getState());
-    const canvas = new CanvasOverlay(new ColorizeCanvas2D(), stateDeps);
+    // const innerCanvas = new ColorizeCanvas2D();
+    // const canvas = new CanvasOverlay(innerCanvas, stateDeps);
```

Comment:
Some messy changes while this is still in prototyping phase. This branch won't merge directly into `main` yet until I fix the old canvas behavior.

Reply:
is this where you will ask the dataset if it is 3d or not?  why not implement the branching here?

Reply:
Yup, there will definitely be branching, though I'm not sure if it belongs here? At the point in which this callback fires, the Dataset is not yet loaded, so we won't know until later if we need the 3D or 2D canvas.

Ideally, we might instantiate just the `CanvasOverlay` here, and have it do an internal switch between the two canvas implementations based on the type of the loaded dataset.